fre/pp/configure_script_yaml.py

```diff
@@ -1,71 +1,84 @@
-import yaml
+from typing import Tuple
 import metomi.rose.config
 
 import fre.yamltools.combine_yamls_script as cy
 
-import logging
 fre_logger = logging.getLogger(__name__)
 
-####################
-def yaml_load(yamlfile):
-    """
-    Load the given yaml.
-    """
-    # Load the main yaml
-    with open(yamlfile,'r') as f:
-        y=yaml.safe_load(f)
-
-    return y
-
 ######VALIDATE#####
-def validate_yaml(yamlfile):
+def validate_yaml(yamlfile: dict) -> None:
     """
-     Using the schema.json file, the yaml format is validated.
+    Validate the format of the yaml file based
+    on the schema.json in gfdl_msd_schemas
+
+    :param yamlfile: Model, settings, pp, and analysis yaml
+                     information combined into a dictionary
+    :type yamlfile: dict
+    :raises ValueError:
+        - if gfdl_mdf_schema path is not valid
+        - combined yaml is not valid
+        - unclear error in validation
+    :return: None
+    :rtype: None
     """
     schema_dir = Path(__file__).resolve().parents[1]
     schema_path = os.path.join(schema_dir, 'gfdl_msd_schemas', 'FRE', 'fre_pp.json')
-    fre_logger.info(f"Using yaml schema '{schema_path}'")
+    fre_logger.info("Using yaml schema '%s'", schema_path)
     # Load the json schema: .load() (vs .loads()) reads and parses the json in one)
     try:
-        with open(schema_path,'r') as s:
+        with open(schema_path,'r', encoding='utf-8') as s:
             schema = json.load(s)
     except:
-        fre_logger.error(f"Schema '{schema_path}' is not valid. Contact the FRE team.")
+        fre_logger.error("Schema '%s' is not valid. Contact the FRE team.", schema_path)
         raise
 
     # Validate yaml
     # If the yaml is not valid, the schema validation will raise errors and exit
     try:
-        validate(instance=yamlfile,schema=schema)
+        validate(instance = yamlfile,schema=schema)
         fre_logger.info("Combined yaml valid")
-    except SchemaError:
-        fre_logger.error(f"Schema '{schema_path}' is not valid. Contact the FRE team.")
-        raise
-    except ValidationError:
-        fre_logger.error("Combined yaml is not valid. Please fix the errors and try again.")
-        raise
-    except:
-        fre_logger.error("Unclear error from validation. Please try to find the error and try again.")
-        raise
+    except SchemaError as exc:
+        raise ValueError(f"Schema '{schema_path}' is not valid. Contact the FRE team.") from exc
+    except ValidationError as exc:
+        raise ValueError("Combined yaml is not valid. Please fix the errors and try again.") from exc
+    except Exception as exc:
+        raise ValueError("Unclear error from validation. Please try to find the error and try again.") from exc
 
 ####################
-def rose_init(experiment,platform,target):
+def rose_init(experiment: str, platform: str, target: str) -> Tuple[metomi.rose.config.ConfigNode, metomi.rose.config.ConfigNode, metomi.rose.config.ConfigNode]:
```

Comment:
I want to say Tuple is now not recommended, and they recommend using `tuple` which doesn't need to be imported 
https://docs.python.org/3/library/stdtypes.html#tuple